### PR TITLE
Authentication centralized and emulated

### PR DESCRIPTION
**Pull request by whitewind664**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Thanks man this really needed doing :)

File: app/src/androidTest/java/com/github/gogetters/letsgo/matchmaking/MatchmakingAndroidTest.kt [hunk 1]
```diff
@@ -22,11 +22,6 @@ public class MatchmakingAndroidTest: EmulatedFirebaseTest() {
 
     @Test
     fun findMatchWorks() {
```
Comment (kprotopapas) [inline app/src/androidTest/java/com/github/gogetters/letsgo/matchmaking/MatchmakingAndroidTest.kt:24]: maybe just remove the whole test?

File: app/src/main/java/com/github/gogetters/letsgo/database/Authentication.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.github.gogetters.letsgo.database
+
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+
+class Authentication {
```
Comment (kprotopapas) [inline app/src/main/java/com/github/gogetters/letsgo/database/Authentication.kt:7]: Is it not better to make it implement an interface so we can mock it? Maybe too late thought

**Review by kprotopapas — APPROVED** [review]
Good overall, but testing is a problem? not sure if it really is just seems to me

File: app/src/main/java/com/github/gogetters/letsgo/database/Authentication.kt (hunk 2 above)
Comment (ghost) [reply app/src/main/java/com/github/gogetters/letsgo/database/Authentication.kt:7]: I believe sticking to Emulating everything is better

**Pull request merged**